records_mover/url/optimizer/__init__.py

```diff
@@ -0,0 +1,232 @@
+import json
+import time
+from contextlib import contextmanager
+from records_mover.url.base import BaseDirectoryUrl
+import logging
+from typing import Iterator, Tuple, Union, Optional, TYPE_CHECKING
+if TYPE_CHECKING:
+    from googleapiclient.discovery import _TransferJobConfig
+    import google.auth.credentials
+    from records_mover.url.gcs.gcs_directory_url import GCSDirectoryUrl
+    from records_mover.url.s3.s3_directory_url import S3DirectoryUrl
+
+
+logger = logging.getLogger(__name__)
+
+
+class CopyOptimizer:
+    # TODO: Figure out where this should be called from
+    # TODO: Document this class
+
+    def copy(self,
+             loc: BaseDirectoryUrl,
+             other_loc: BaseDirectoryUrl) -> bool:
+        # TODO: Implement this and swap out call
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker